src/rpc/util.h

```diff
@@ -383,6 +400,36 @@ class RPCHelpMan
     RPCHelpMan(std::string name, std::string description, std::vector<RPCArg> args, RPCResults results, RPCExamples examples, RPCMethodImpl fun);
 
     UniValue HandleRequest(const JSONRPCRequest& request) const;
+    /**
+     * Helper to get a request argument or its default value.
+     * This function only works during m_fun(), i.e. it should only be used in RPC method implementations.
+     * Use Arg<Type> to get the argument or its default value.
+     * Use Arg<Type*> to get the argument or a falsy value.
+     */
+    template <typename R>
+    auto Arg(size_t i) const
+    {
+        if constexpr (std::is_pointer_v<R>) {
+            // Return optional argument (without default).
+            using U = std::remove_pointer_t<R>;
+            if constexpr (std::is_integral_v<U> || std::is_floating_point_v<U>) {
+                // Return numbers by value, wrapped in optional.
+                return ArgValue<std::optional<U>>(i);
```

Comment:
I'd be on board with that approach, too. A bit more explicit and reduces implementation complexity, while I think usability is as good, since we already require the user to distinguish between the arg having a default or not.

nit/bikeshed: I'd prefer `Arg` over `ArgOrDefault`.